### ISSUE TITLE
Add additional config to skip violation check when the changes are less than a threshold

### ISSUE THREAD

**Issue by praveen-kumar85** [SurpSG/diff-coverage-maven-plugin#40]
When there are fewer changes, very often we don't cover all the possible instructions for these changes in the test cases. And in such cases, we should have the option to avoid the violation error by defining the threshold for fewer changes (number of instructions, lines, branches, etc.)

**Comment by SurpSG** [comment]
Hi @praveen-kumar85 

## Workaround

I solved the issue you discribed in the next way:
1. Declare a property in pom.xml:
```xml
...
    <properties>
        <failOnDiffCoverageViolation>true</failOnDiffCoverageViolation> <!-- fail by default -->
    </properties>
...
```
2. Setup the plugin to use the property:   
```xml
            <plugin>
                <groupId>com.github.surpsg</groupId>
                <artifactId>diff-coverage-maven-plugin</artifactId>
                <version>0.3.2</version>
                <configuration>
                    ...
                    <violations>
                        <failOnViolation>${failOnDiffCoverageViolation}</failOnViolation>  <!-- consume the property -->
                        <minCoverage>0.9</minCoverage>
                    </violations>
                </configuration>
                ...
            </plugin>
```
3. Then invoke the plugin with overriding property value 
```bash
mvn install -DfailOnDiffCoverageViolation=false
````

----

## CI usage

### Github Actions
Suppose you are using Github Actions:
1. Create a custom [label](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) (let's say `ignore-diff-coverage`). 
2. Update your github action:

```yaml
      - name: Build with Maven
        run: mvnw install -DfailOnDiffCoverageViolation=${{ !contains(github.event.pull_request.labels.*.name, 'ignore-diff-coverage') }}
```

OR

```yaml
# in this case failOnDiffCoverageViolation maven property is not used for passing to maven 
# but it still would be handy for local run

      - name: Build with Maven
        continue-on-error: ${{ contains(github.event.pull_request.labels.*.name, 'ignore-diff-coverage') }} 
        run: mvnw install
```
 
3. Apply the label to your pull request that requires skipping of diff coverage check

### Another CIs
You can apply the similar approach to Gitlab or Jenkins piplines for sure. Also, I belive you can use the approach for CIs like Travis or Circle.

### Real example
* Pull request: https://github.com/SurpSG/diff-coverage-maven-plugin/pull/41 with label `ignore-diff-coverage`
* [Logs](https://github.com/SurpSG/diff-coverage-maven-plugin/actions/runs/5314959558/jobs/9622776145#step:5:128) where we can see that diff coverage was suppressed but [by default](https://github.com/SurpSG/diff-coverage-maven-plugin/pull/41/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R63) the plugin fails a build if coverage violated. 

### Summary
Advantuges: 
- You intetionally suppress diff coverage failure. Small change in code doesn't mean that the coverage check should be omitted. So you can decide whether ignore diff coverage check per particular PR.

Disadvantuges:
- You must manually suppress diff coverage failure if required 🙃

Yeah, it depends on your strategy.

**Comment by SurpSG** [comment]
If the approach above doesn't suitable for you then await few days(up to 2 weeks) while I implement the feature you ask.

**Comment by praveen-kumar85** [comment]
@SurpSG thanks a lot for the response. This is quite useful for manually adjusting the failure setting without having to touch the code. For our current use cases, it is good enough. But still, I think we should have a configurable parameter to skip the violations just the way how sonar does it.

**Comment by SurpSG** [comment]
@praveen-kumar85 

Going to implement the feature you requested. Here example how it could be configured:

https://github.com/SurpSG/diff-coverage-maven-plugin/pull/50/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R62

I would be pleased to receive feedback

**Comment by SurpSG** [comment]
@DaScheid you are right, I've lost that changes during migration to newer version of delta coverage core dependency.
Now the feature is available in version [1.2.0](https://github.com/gw-kit/delta-coverage-maven-plugin/releases/tag/v1.2.0)

**Comment by DaScheid** [comment]
Thank you for adding this feature

As a first feedback:
1. it works as expected, e.g. if threshold is 10 and only 9 lines are changed then no line-coverage is required
2. it could be useful to have a more granular configuration, e.g. to configure 10 lines without coverage, but only 5 branches without coverage - as more branches could indicate that more complex logic was added with a code-change. More changed lines without coverage could be desired if it's only e.g. some logging

**Comment by SurpSG** [comment]
> Thank you for adding this feature
> 
> As a first feedback:
> 
> 1. it works as expected, e.g. if threshold is 10 and only 9 lines are changed then no line-coverage is required
> 2. it could be useful to have a more granular configuration, e.g. to configure 10 lines without coverage, but only 5 branches without coverage - as more branches could indicate that more complex logic was added with a code-change. More changed lines without coverage could be desired if it's only e.g. some logging

Pls, create a new issue(feature request)
